### ISSUE TITLE
Create a stub installer for Windows w/ Omaha

### ISSUE THREAD

**Issue by bbondy** [brave/brave-browser#65]
Use NSIS to create a GUI installer similar to mozilla Firfox's stub installer

**Comment by bbondy** [comment]
So the goal is to create a GUI for our Windows installer which is currently just a silent installer. It will be in the form of a small download which gets the latest build from a remote endpoint.

mozilla Firefox source code can be searched and browsed here:
https://dxr.mozilla.org/mozilla-central/

You will probably want to clone the source code yourself from Mozilla Firefox:
```
hg clone https://hg.mozilla.org/mozilla-central/ firefox
cd firefox
```
You probably don't have to build it, but if you want to:
https://developer.mozilla.org/en-US/docs/Mozilla/Developer_guide/Build_Instructions/Windows_Prerequisites

NSIS can be downloaded here:
http://nsis.sourceforge.net/Download

A simple tutorial you should read just to get an idea of how things work and make a basic simple installer is here:
http://nsis.sourceforge.net/Simple_tutorials

NSIS files have a file extension of `.nsi`, and for include files `.nsh`.  Plugins for nsis are compiled c++ code, mozilla just includes the pre-compiled.

Installers are made by running ` makensisw.exe` with input of an `.nsi` file.

The NSIS script at Firefox for the stub installer is here:
https://dxr.mozilla.org/mozilla-central/source/browser/installer/windows/nsis/stub.nsi

Some include files are included with NSIS, others are in the mozilla tree mostly copied from commonly used scripts on the internet.  Other than `browser/installer/windows/nsis` this file's directory is also used `toolkit/mozapps/installer/windows/nsis/common.nsh`

You can just keep the same copyright in the copied files since they also use the MPL-2.0 license.

Once you know the artwork dimensions, take a screenshot of how they look and do a DM with James, Brad and me about it.  I can help try to expedite getting artwork for it.

Any questions or problems please let me know, I unfortunately had to spend a lot of time working with NSIS in the past :).

**Comment by RyanJarv** [comment]
Haven't looked into this much but I believe this is what mini_installer is supposed to do in the chromium source but for whatever reason (guessing building with official_build=false) this is the main executable in our builds.

@bbondy

**Comment by bbondy** [comment]
I thought that's what we used for the silent installer

**Comment by RyanJarv** [comment]
It is, originally I suspected there is something we're missing from the code at https://github.com/chromium/chromium/tree/master/chrome/installer/mac/app but didn't realize that was just OSX before (which we likely can't use with sparkle).

But looking at https://github.com/chromium/chromium/blob/master/chrome/installer/mini_installer/mini_installer.cc#L5 seems mini_installer should be about ~5kb's.

**Comment by bbondy** [comment]
@RyanJarv I hope I can be proved wrong but I think it's the same as I did previously which just gives the silent installer which gets renamed here:
https://github.com/brave/brave-core/blob/master/build/win/BUILD.gn#L30

Worth looking into @emerick before starting down the mozilla track that I outlined.

**Comment by emerick** [comment]
@bbondy Definitely, I'll investigate.

**Comment by bbondy** [comment]
@emerick 
Change of plans, @simonhong knows how to do this with Omaha, so he's going to own it.  We'll get you a different task to work on.

**Comment by emerick** [comment]
@bbondy OK, no worries.

**Comment by simonhong** [comment]
Omaha has it's own stub install UI that shows download progress and install progress.
It is similar to chrome's installer UI.
I think we can modify omaha's UI for our UI.

**Comment by RyanJarv** [comment]
Closing this in favor of https://github.com/brave/brave-browser/issues/414
